l1/l1.go

```diff
@@ -78,98 +77,127 @@ func (c *Client) checkChainID(ctx context.Context) error {
+func (c *Client) Run(ctx context.Context) error { //nolint:gocyclo
 	defer c.l1.Close()
 	if err := c.checkChainID(ctx); err != nil {
 		return err
 	}
 
 	buffer := 128
 
-	logStateUpdateChan := make(chan *contract.StarknetLogStateUpdate, buffer)
-	subUpdates, err := c.subscribeToUpdates(ctx, logStateUpdateChan)
+	updateChan := make(chan *contract.StarknetLogStateUpdate, buffer)
+	updateSub, err := c.subscribeToUpdates(ctx, updateChan)
 	if err != nil {
 		return err
 	}
-	defer subUpdates.Unsubscribe()
+	defer updateSub.Unsubscribe()
 
 	headerChan := make(chan *types.Header, buffer)
-	subHeaders, err := c.subscribeToHeaders(ctx, headerChan)
+	headerSub, err := c.subscribeToHeaders(ctx, headerChan)
 	if err != nil {
 		return err
 	}
-	defer subHeaders.Unsubscribe()
+	defer headerSub.Unsubscribe()
 
 	for {
 		select {
 		case <-ctx.Done():
 			return nil
-		case err := <-subHeaders.Err():
+		case err := <-headerSub.Err():
 			c.log.Warnw("L1 header subscription failed, resubscribing", "error", err)
-			subHeaders.Unsubscribe()
+			headerSub.Unsubscribe()
 
-			subHeaders, err = c.subscribeToHeaders(ctx, headerChan)
+			headerSub, err = c.subscribeToHeaders(ctx, headerChan)
 			if err != nil {
 				return err
 			}
-			defer subHeaders.Unsubscribe() //nolint:gocritic
-		case header := <-headerChan:
-			l1Height := header.Number.Uint64()
-			c.log.Debugw("Received L1 header", "number", l1Height, "hash", header.Hash().Hex())
+			defer headerSub.Unsubscribe() //nolint:gocritic
+		case <-headerChan:
 		Outer:
-			// Check for updates.
-			// We need to loop in case there were multiple LogStateUpdates emitted.
 			for {
 				select {
-				case err := <-subUpdates.Err():
+				case err := <-updateSub.Err():
+					// TODO can we use geth's event.Resubscribe?
 					c.log.Warnw("L1 update subscription failed, resubscribing", "error", err)
-					subUpdates.Unsubscribe()
+					updateSub.Unsubscribe()
 
-					subUpdates, err = c.subscribeToUpdates(ctx, logStateUpdateChan)
+					updateSub, err = c.subscribeToUpdates(ctx, updateChan)
 					if err != nil {
 						return err
 					}
-					defer subUpdates.Unsubscribe() //nolint:gocritic
-				case logStateUpdate := <-logStateUpdateChan:
+					defer updateSub.Unsubscribe() //nolint:gocritic
+				case logStateUpdate := <-updateChan:
 					c.log.Debugw("Received L1 LogStateUpdate",
 						"number", logStateUpdate.BlockNumber,
 						"stateRoot", logStateUpdate.GlobalRoot.Text(felt.Base16),
 						"blockHash", logStateUpdate.BlockHash.Text(felt.Base16))
+
 					if logStateUpdate.Raw.Removed {
-						// NOTE: we only modify the local confirmationQueue upon receiving reorged logs.
-						// We assume new logs will soon follow, so we don't notify the l2Chain.
-						c.confirmationQueue.Reorg(logStateUpdate.Raw.BlockNumber)
+						for l1BlockNumber := range c.nonFinalisedLogs {
+							if l1BlockNumber <= logStateUpdate.Raw.BlockNumber {
+								delete(c.nonFinalisedLogs, l1BlockNumber)
+							}
```

Comment:
```suggestion
							if l1BlockNumber > logStateUpdate.Raw.BlockNumber {
								delete(c.nonFinalisedLogs, l1BlockNumber)
							}
```

Shouldn't it be the above? I don't exactly understand why we would want to delete blocks lower than the current height? We should delete it and anything above, right?